nwb_conversion_tools/tools/neo/neo.py

```diff
@@ -0,0 +1,608 @@
+"""Author: Luiz Tauffer."""
+from typing import Optional, Tuple
+import distutils.version
+import uuid
+from datetime import datetime
+from pathlib import Path
+import warnings
+
+import neo.io.baseio
+import pynwb
+
+from ..nwb_helpers import add_device_from_metadata
+from ...utils import OptionalFilePathType
+
+
+response_classes = dict(
+    voltage_clamp=pynwb.icephys.VoltageClampSeries,
+    current_clamp=pynwb.icephys.CurrentClampSeries,
+    izero=pynwb.icephys.IZeroClampSeries,
+)
+
+stim_classes = dict(
+    voltage_clamp=pynwb.icephys.VoltageClampStimulusSeries,
+    current_clamp=pynwb.icephys.CurrentClampStimulusSeries,
+)
+
+
+# TODO - get electrodes metadata
+def get_electrodes_metadata(neo_reader, electrodes_ids: list, block: int = 0) -> list:
+    """
+    Get electrodes metadata from Neo reader.
+
+    The typical information we look for is the information accepted by pynwb.icephys.IntracellularElectrode:
+      - name – the name of this electrode
+      - device – the device that was used to record from this electrode
+      - description – Recording description, description of electrode (e.g., whole-cell, sharp, etc)
+      - comment: Free-form text (can be from Methods)
+      - slice – Information about slice used for recording.
+      - seal – Information about seal used for recording.
+      - location – Area, layer, comments on estimation, stereotaxis coordinates (if in vivo, etc).
+      - resistance – Electrode resistance COMMENT: unit: Ohm.
+      - filtering – Electrode specific filtering.
+      - initial_access_resistance – Initial access resistance.
+
+    Parameters
+    ----------
+    neo_reader ([type]): Neo reader
+    electrodes_ids (list): List of electrodes ids.
+    block (int, optional): Block id. Defaults to 0.
+
+    Returns
+    -------
+    list: List of dictionaries containing electrodes metadata.
+    """
+    return []
+
+
+def get_number_of_electrodes(neo_reader) -> int:
+    """Get number of electrodes from Neo reader."""
+    # TODO - take in account the case with multiple streams.
+    return len(neo_reader.header["signal_channels"])
+
+
+def get_number_of_segments(neo_reader, block: int = 0) -> int:
+    """Get number of segments from Neo reader."""
+    return neo_reader.header["nb_segment"][block]
+
+
+def get_command_traces(neo_reader, block: int = 0, segment: int = 0, cmd_channel: int = 0) -> Tuple[list, str, str]:
+    """
+    Get command traces (e.g. voltage clamp command traces).
+
+    Parameters
+    ----------
+    neo_reader : neo.io.baseio
+    block : int, optional.
+        Defaults to 0.
+    segment : int, optional
+        Defaults to 0.
+    cmd_channel : int, optional
+        ABF command channel (0 to 7). Defaults to 0.
+    """
+    try:
+        traces, titles, units = neo_reader.read_raw_protocol()
+        return traces[segment][cmd_channel], titles[segment][cmd_channel], units[segment][cmd_channel]
+    except Exception as e:
```

Comment:
Does the resulting error here happen to be anything more specific? Like a `NotImplementedError` or something? Generally try to avoid blanket excepts for things like this.

Reply:
well, this is not really a `NotImplementedError`, I don't know which built-in exception would characterize it better. What would you suggest?
We have the message there though, that should be enough for informative error logging

Reply:
@luiztauffer can you explain why the try clause would fail? In general I try to avoid the try/except pattern. Sometimes it is the best way to go, but it at least needs good comments explaining why